include/zephyr/drivers/bluetooth/hci_driver_bluenrg.h

```diff
@@ -0,0 +1,47 @@
+/** @file
+ *  @brief BlueNRG HCI extended API.
+ */
+
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BLUETOOTH_HCI_DRIVER_BLUENRG_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BLUETOOTH_HCI_DRIVER_BLUENRG_H_
+
+/**
+ * @brief BlueNRG HCI Driver-Specific API
+ * @defgroup bluenrg_hci_driver BlueNRG HCI driver extended API
+ * @ingroup bluetooth
+ * @{
+ */
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Hardware reset the BlueNRG network coprocessor.
+ *
+ * Performs hardware reset of the BLE network coprocessor.
+ * It can also force to enter firmware updater mode.
+ *
+ * @param updater_mode flag to indicate whether updater mode needs to be entered.
+ *
+ * @return a non-negative value indicating success, or a
+ * negative error code for failure
+ */
+
+int bluenrg_bt_reset(bool updater_mode);
```

Comment:
Note: along with #72323 this will likely need to become some kind of HCI driver API extension, e.g. taking a `const struct device *dev` parameter.

Reply:
Instead of making this a direct C API, did you consider just passing the vendor HCI command to the driver and let the driver parse the command and initiate this procedure?

Reply:
Actually, the proposed extended API is not implemented by vendor HCI commands, but it is an API to perform hardware reset (i.e. pin reset) and bootloader activation of ST products using combination of GPIO pins.
Indeed, the same tasks can be performed by some standard or vendor specific commands, like:
- reset: HCI_RESET
- bootloader activation: ACI_HAL_UPDATER_START
 
These HCI commands require the device to be able to listen to the received commands and there are cases where this is not possible (e.g. BlueNRG firmware crash or missing). The GPIO activation provides an alternative way to perform the same task without relying on the device firmware.
 
The usage of this new API is triggered by a special UART sequences which are not HCI vendor specific commands, but they are interpreted by the sample application and in turn translated to call bluenrg_bt_reset API when it's appropriate.
 
For connection to different hosts, like BlueZ still the hci_uart sample is the way to go.